rust/lance/src/dataset.rs

```diff
@@ -1763,6 +1763,18 @@ impl Dataset {
         Ok(())
     }
 
+    /// Get config items based on the given keys.
+    pub async fn get_configs(&mut self, keys: &[&str]) -> Result<HashMap<String, String>> {
```

Comment:
what if I want to get all the configs in the table? Can we make this `keys: Option<&[&str]>`?

Reply:
oh I see, I totally missed that `config` API. In that case why do we want another API? Why not just get the whole map and then filter it based on the keys?

Reply:
> IMO, the most common scenario is to get values via keys from a config hash map. 

Yes totally agree

> then I need to iterate the hash map, which is not very elegant.

Yeah I see there is another update to that API in #4039. If the main issue is retrieval through iteration, why not just have 

```
pub fn config(&self) -> HashMap<String, String>
```

and then you can either use the whole thing, or retrieve it by key through the map. 

This is not big metadata, and mainly used in only non-query path, so I feel we don't need to be super conservative on saving the map construction memory or perf. And in #4039 we are changing it to copy the key value string anyway.

Reply:
Sounds reasonable, @majin1102 WDYT about jack's idea?

Reply:
For concision I agree with Jack